File: api/core/src/main/kotlin/org/chatRoom/core/model/Session.kt
package org.chatRoom.core.model

import kotlinx.serialization.Serializable
import org.chatRoom.core.serializer.OffsetDateTimeSerializer
import org.chatRoom.core.aggreagte.Session as SessionAggregate
import org.chatRoom.core.valueObject.Id
import org.chatRoom.core.valueObject.Token
import java.time.OffsetDateTime
import java.time.ZoneOffset

@Serializable
data class Session(
    val id: Id,
    val userId: Id,
    val token: Token,
    @Serializable(with = OffsetDateTimeSerializer::class)
    val dateValidUntil: OffsetDateTime,
    @Serializable(with = OffsetDateTimeSerializer::class)
    val dateCreated: OffsetDateTime,
) {
    constructor(session: SessionAggregate) : this(
        id = session.modelId,
        userId = session.userId,
        token = session.token,
        dateValidUntil = session.dateValidUntil.atOffset(ZoneOffset.UTC),
        dateCreated = session.dateCreated.atOffset(ZoneOffset.UTC),
    )
}

File: web/models/session.ts
export default class Session {
    id: string
    userId: string
    secret: string
    token: string
    dateValidUntil: Date
    dateCreated: Date

    constructor(rawData: any) {
        this.id = rawData.id
        this.userId = rawData.userId
        this.secret = rawData.secret
        this.token = rawData.token
        this.dateValidUntil = new Date(rawData.dateValidUntil)
        this.dateCreated = new Date(rawData.dateCreated)
    }
}
